src/switch_core_session.c

```diff
@@ -2743,9 +2743,13 @@ SWITCH_DECLARE(switch_status_t) switch_core_session_execute_application_get_flag
 	}
 
 	if ((application_interface = switch_loadable_module_get_application_interface(app)) == 0) {
-		switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_ERROR, "Invalid Application %s\n", app);
-		switch_channel_hangup(session->channel, SWITCH_CAUSE_DESTINATION_OUT_OF_ORDER);
-		switch_goto_status(SWITCH_STATUS_FALSE, done);
+		if (strcmp(app, "spandsp_start_dtmf") && switch_channel_get_variable(session->channel, "inbound_bypass_media") && switch_channel_get_variable(session->channel, "ignore_spandsp_error")){
```

Comment:
strcmp() returns nonzero when strings differ and 0 on match, so this should be
`!strcmp(app, "spandsp_start_dtmf"`

If this works as you intended then line 2747 would get executed and then code would continue at line 2755, but at that line it would crash because `application_interface` is NULL.
To fix this at line 2748 you need to return.

These bugs are very basic mistakes and you would catch them if you made any tests of your code changes. **Please always run some tests, before making a PR**. Bugs happen always, and even myself, when making even slightest changes I have to run at least one, two basic tests to make sure it compiles, do not crash, etc.